torch/nativert/executor/Placement.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <c10/core/Device.h>
+#include <c10/util/Logging.h>
+
+#include <optional>
+#include <unordered_map>
+
+namespace torch::nativert {
+
+/**
+ * @brief Normalizes a c10::Device by ensuring a canonical form.
```

Comment:
nit: a "canonical form" for Device is not a thing within PyTorch. So you most likely want to use another word here.